src/main/java/dinkplugin/message/DiscordMessageHandler.java

```diff
@@ -162,6 +167,36 @@ private void sendMessage(HttpUrl url, MultipartBody.Builder requestBody, int att
         });
     }
 
+    private static CompletableFuture<byte[]> captureScreenshot(DrawManager drawManager, double scalePercent) {
+        CompletableFuture<Image> future = new CompletableFuture<>();
+        drawManager.requestNextFrameListener(future::complete);
+        return future.thenApply(ImageUtil::bufferedImageFromImage)
+            .thenApply(input -> Utils.rescale(input, scalePercent))
+            .thenApply(image -> {
+                try {
+                    return Utils.convertImageToByteArray(image);
+                } catch (IOException e) {
+                    throw new CompletionException("Could not convert image to byte array", e);
+                }
+            })
+            .thenApply(bytes -> {
+                int n = bytes.length;
+                if (n <= Embed.MAX_IMAGE_SIZE)
+                    return bytes; // already compliant; no further rescale necessary
+
+                // calculate scale factor to comply with MAX_IMAGE_SIZE
```

Comment:
With the automatic rescale logic, should we still bother exposing an advanced setting for a specific rescale % ?